invokeai/app/services/model_install/model_install_default.py

```diff
@@ -238,6 +244,99 @@ def scan_directory(self, scan_dir: Path, install: bool = False) -> List[str]:  #
+        self.record_store.del_model(key)
+
+    def delete(self, key: str) -> None:  # noqa D102
+        """Unregister the model. Delete its files only if they are within our models directory."""
+        model = self.record_store.get_model(key)
+        models_dir = self.app_config.models_path
+        model_path = models_dir / model.path
+        if model_path.is_relative_to(models_dir):
+            self.unconditionally_delete(key)
+        else:
+            self.unregister(key)
+
+    def unconditionally_delete(self, key: str) -> None:  # noqa D102
+        model = self.record_store.get_model(key)
+        path = self.app_config.models_path / model.path
+        if path.is_dir():
+            rmtree(path)
+        else:
+            path.unlink()
+        self.unregister(key)
+
+    # --------------------------------------------------------------------------------------------
+    # Internal functions that manage the installer threads
+    # --------------------------------------------------------------------------------------------
+    def _start_installer_thread(self) -> None:
+        threading.Thread(target=self._install_next_item, daemon=True).start()
+        self._running = True
+
+    def _install_next_item(self) -> None:
+        done = False
+        while not done:
+            if self._stop_event.is_set():
+                done = True
+                continue
+            try:
+                job = self._install_queue.get(timeout=1)
+            except Empty:
+                continue
+
+            assert job.local_path is not None
+            try:
+                if job.cancelled:
+                    self._signal_job_cancelled(job)
+                elif job.errored:
+                    self._signal_job_errored(job)
+                elif (
+                    job.waiting or job.downloading
+                ):  # local jobs will be in waiting state, remote jobs will be downloading state
+                    self._signal_job_running(job)
+                    if job.inplace:
+                        key = self.register_path(job.local_path, job.config_in)
+                    else:
+                        key = self.install_path(job.local_path, job.config_in)
+                    job.config_out = self.record_store.get_model(key)
+
+                    # enter the metadata, if there is any
+                    if metadata := self._metadata_cache.get(job.source):
+                        self._metadata_store.add_metadata(key, metadata)
+                    self._signal_job_completed(job)
+
+            except InvalidModelConfigException as excp:
+                if "unrecognized suffix" in str(excp):
+                    job.set_error(
+                        InvalidModelConfigException(
+                            f"The file {job.local_path} is not recognized as a model. Check the URL and access token."
+                        )
+                    )
+                    self._signal_job_errored(job)
```

Comment:
The code was not right and I've fixed. The intent is always to fail on this exception but to change the exception error message for this specific case to indicate that there might be a problem with the access token. Unfortunately when downloading from Civitai, if an access token is needed and not provided, the request returns the HTML of the login page rather than the model itself, and this generates the unrecognized suffix failure.

Reply:
Gotcha. Could we detect if there's a redirect and handle that accordingly?

Hmm - if I recall correctly, there is *always* a redirect to their cloud storage, so maybe that's not gonna work.

Reply:
I can see if interrogating the `content-type` field helps. It will probably be `text/html` for the login page, and `binary/octet-something` for the model files. I can intercept a bad content type early in the process and issue a better error message.

Reply:
I've set things up so that the downloader sets a content_type field early on in the request process. This allows the installer to issue a more informative error message on an `InvalidModelConfigException` without relying on parsing the exception message.